From 5bf0c6dd844da260796bebc79cafddffe1fedb33 Mon Sep 17 00:00:00 2001
From: Guo Jie <guojie@loongson.cn>
Date: Wed, 15 Feb 2023 17:08:09 +0800
Subject: [PATCH 27/28] Disable stackrealign when ISA_HAS_LASX is false to
 avoid linux-kernel building warning 'objtool: XXX: return with modified stack
 frame'.

Change-Id: I9bf5334068b5db16681149edc0c27f01e81939d5
---
 src/gcc/config/loongarch/loongarch.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/gcc/config/loongarch/loongarch.c b/src/gcc/config/loongarch/loongarch.c
index 81100d754..c56865fc2 100644
--- a/src/gcc/config/loongarch/loongarch.c
+++ b/src/gcc/config/loongarch/loongarch.c
@@ -7127,6 +7127,8 @@ loongarch_option_override_internal (struct gcc_options *opts,
       inform (input_location,
 	      "valid arguments to %<-mveclibabi=%> are: %s", "sleef");
     }
+  if (!ISA_HAS_LASX)
+    loongarch_stack_realign = 0;
 }
 
 
-- 
2.31.1

